applications/stepswitch/src/stepswitch_util.erl

```diff
@@ -91,41 +101,58 @@ correct_shortdial(<<"+", Number/binary>>, CIDNum, DeniedCallRestrictions) ->
 correct_shortdial(Number, <<"+", CIDNum/binary>>, DeniedCallRestrictions) ->
     correct_shortdial(Number, CIDNum, DeniedCallRestrictions);
 correct_shortdial(Number, CIDNum, DeniedCallRestrictions) when is_binary(CIDNum) ->
+    correct_shortdial(Number, CIDNum, DeniedCallRestrictions, should_correct_shortdial(Number)).
+
+-spec correct_shortdial(kz_term:ne_binary(), kz_term:ne_binary(), kz_json:object(), {boolean(), pos_integer()}) -> kz_term:api_ne_binary().
+correct_shortdial(Number, CIDNum, DeniedCallRestrictions, {'true', _Length}) ->
     case shortdial_correction_length(Number, CIDNum) of
         0 ->
             lager:debug("unable to correct shortdial ~s via CID ~s"
                        ,[Number, CIDNum]
                        ),
             'undefined';
         Length ->
-            try_to_correct(Number, CIDNum, DeniedCallRestrictions, Length)
-    end.
-
--spec try_to_correct(kz_term:ne_binary(), kz_term:ne_binary(), kz_json:object(), non_neg_integer()) -> kz_term:api_ne_binary().
-try_to_correct(Number, CIDNum, DeniedCallRestrictions, Length) ->
+            maybe_deny_reclassified_number(do_correct_shortdial(Number, CIDNum, Length)
+                                          ,CIDNum
+                                          ,DeniedCallRestrictions
+                                          )
+    end;
+correct_shortdial(Number, _CIDNum, _DeniedCallRestrictions, {'false', Length}) ->
+    lager:debug("~s's lenght (~p) doesn't match shortdial_correction_if_length_is param's value (~p), skipping"
+               ,[Number, byte_size(Number), Length]
+               ),
+    'undefined'.
+
+-spec should_correct_shortdial(kz_term:ne_binary()) -> {boolean(), pos_integer()}.
+should_correct_shortdial(Number) ->
+    Length = kapps_config:get_integer(?SS_CONFIG_CAT, <<"shortdial_correction_if_length_is">>, 0),
+    %% The length is also included on the response just to avoid having to read this configuration
+    %% parameter again when printing the debug line. So it is only useful for the `{false, N}' branch.
+    {0 =:= Length
```

Comment:
I would prefer the logic portion (comparing Length and Number) in its own function:

```erlang
should_correct_shortdial(Number) ->
    Length = kapps_config:(...),
    should_correct_shortdial(Number, Length).

should_correct_shortdial(_Number, 0) -> {'true', 0};
should_correct_shortdial(Number, Length) ->
    {byte_size(Number) =:= Length, Length}.
```

This is also easily testable without using meck or other means.